src/meltano/cli/job.py

```diff
@@ -278,8 +303,10 @@ def _validate_tasks(project: Project, task_set: TaskSets) -> bool:
             block_parser = BlockParser(logger, project, blocks)
             parsed_blocks = list(block_parser.find_blocks(0))
```

Comment:
So what would be interesting here is if we attached a [PluginsContent](https://github.com/meltano/meltano/blob/main/src/meltano/core/tracking/plugins.py#L88) to tracker. The PluginsContext basically attaches info to every event that follows with a dict describing what plugins are in use.

Normally `PluginsContext()` wants a `tuple[ProjectPlugin, str]` but what you get from the block parser is a list of `blocks`. So I have:

https://github.com/meltano/meltano/blob/c41c2c153c00c0e01b8eabe5160429de6d6683b6/src/meltano/core/tracking/plugins.py#L41-L45

Which lets you get the PluginsContext for a `single` block (but here you've got a list of blocks). You could crib from that though to build a PluginsContext with all of the plugins. Something like:


```python
def the_thing(blocks: parsed_blocks: list[BlockSet | PluginCommandBlock]) -> PluginsTrackingContext
    plugins: list[(ProjectPlugin, str)] = []
    for blk in list_of_blocks:
        if isinstance(blk, BlockSet):
                for plugin_block in blk.blocks:
                    plugins.append((plugin_block.context.plugin, plugin_block.plugin_args))
            if isinstance(blk, PluginCommandBlock):
                plugins.append((blk.context.plugin, blk.command))
    return PluginsTrackingContext(plugins)

---

tracker.add_contexts(the_thing( parsed_blocks))

```

Reply:
That's definitely doable. The idea being that we'd have telemetry on what plugins are being scheduled?  I have zero context on Snowplow and am just now getting local events firing to snowplow-micro to take a look at what these events actually look like, but I could definitely see having that granularity of data being useful. Do we currently get block-level/plugin-level data like that from `meltano run` or `meltano schedule` if either of those are running `job`s instead of parsing blocks directly from CLI args?

CC: @pnadolny13 in case you have thoughts on usefulness of knowing individual plugins being added to jobs

Reply:
> The idea being that we'd have telemetry on what plugins are being scheduled? 

Yea exactly, so for example `meltano run` will report events for each block its executing, and as part of that its reporting what plugins are associated with that (even if you're asking it to exec a job)  

Similar behavior for invoke and config , they both attach a plugins context for the plugin thats being interacted with. 

I think for `jobs` this will be handy because it will help us figure out what sorta plugins/run commands are turning into jobs, but you're definitely right @pnadolny13 can call the shots on that and let us know whether or not to include plugin data.